test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+			YPM::s_highIP,
+			"Remove function call from statement block"
+		);
+	}
+);
+
+// Add variable declaration
+static YPR<Block> addVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				_message->add_statements()->set_allocated_decl(new VarDecl());
+				// Hoist var decl to beginning of block
+				if (_message->statements_size() > 1)
+					_message->mutable_statements(0)->Swap(
+						_message->mutable_statements(_message->statements_size() - 1)
+					);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add variable declaration to statement block"
+		);
+	}
+);
+
+// Add multivar decl
+static YPR<Block> addMultiVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				auto decl = new MultiVarDecl();
+				decl->set_num_vars(_rand());
+				_message->add_statements()->set_allocated_multidecl(decl);
+				// Hoist multi var decl to beginning of block
+				if (_message->statements_size() > 1)
+					_message->mutable_statements(0)->Swap(
+						_message->mutable_statements(_message->statements_size() - 1)
+					);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add multi variable declaration to statement block"
+		);
+	}
+);
+
+// Remove variable declaration
+static YPR<Block> removeVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_decl())
+					{
+						delete stmt.release_decl();
+						stmt.clear_decl();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove variable declaration from statement block"
+		);
+	}
+);
+
+// Remove multi variable declaration
+static YPR<Block> removeMultiVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_multidecl())
+					{
+						delete stmt.release_multidecl();
+						stmt.clear_multidecl();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove multi variable declaration from statement block"
+		);
+	}
+);
+
+// Add function definition
+static YPR<Block> addFuncDef(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				auto funcDef = new FunctionDef();
+				funcDef->set_num_input_params(_rand());
+				funcDef->set_num_output_params(_rand());
+				funcDef->set_allocated_block(new Block());
+				// TODO: Add assignments to output params if any
+				_message->add_statements()->set_allocated_funcdef(funcDef);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add function definition to statement block"
+		);
+	}
+);
+
+// Remove function definition
```

Comment:
If a function is removed, are references to it removed as well?

Reply:
Yeah. We have a map of function definitions that is used to reference a function. Function definitions are visited before calls. So if a definition is absent, it won't appear in this map, and consequently won't be referenced.

I need to see if this pre-condition is asserted somewhere :-)